scripts/generate_panelite_content.py

```diff
@@ -14,34 +14,87 @@
+with open(PANEL_BASE/"scripts"/"panelite_dependencies.json", "r", encoding="utf8") as file:
+    DEPENDENCIES = json.load(file)
+DEPENDENCY_NOT_IMPORTABLE = [
+    "streamz", # https://github.com/python-streamz/streamz/issues/467,
+    "vtk", # https://gitlab.kitware.com/vtk/vtk/-/issues/18806
+]
+NOTEBOOK_ISSUES = {
+    "Getting_Started.ipynb": ["https://github.com/holoviz/panel/issues/4416"],
+    "reference/panes/DataFrame.ipynb": ["https://github.com/python-streamz/streamz/issues/467"],
+    "reference/panes/HoloViews.ipynb": ["https://github.com/holoviz/panel/issues/4393"],
+    "reference/panes/IPyWidget.ipynb": ["https://github.com/holoviz/panel/issues/4394", "https://github.com/widgetti/ipyvolume/issues/427"],
+    "reference/panes/Matplotlib.ipynb": ["https://github.com/holoviz/panel/issues/4394"],
+    "reference/panes/Param.ipynb": ["https://github.com/holoviz/panel/issues/4393"],
+    "reference/panes/Reacton.ipynb": ["https://github.com/holoviz/panel/issues/4394"],
+    "reference/panes/Str.ipynb": ["https://github.com/holoviz/panel/issues/4396"],
+    "reference/panes/Streamz.ipynb": ["https://github.com/python-streamz/streamz/issues/467"],
+    "reference/panes/VTK.ipynb": ["https://gitlab.kitware.com/vtk/vtk/-/issues/18806"],
+    "reference/panes/VTKVolume.ipynb": ["https://gitlab.kitware.com/vtk/vtk/-/issues/18806"],
+    "reference/widgets/CrossSelector.ipynb": ["https://github.com/holoviz/panel/issues/4398"],
+    "reference/widgets/Debugger.ipynb": ["https://github.com/holoviz/panel/issues/4399"],
+    "reference/widgets/EditableIntSlider.ipynb": ["https://github.com/holoviz/panel/issues/4400"],
+    "reference/widgets/FileDownload.ipynb": ["https://github.com/holoviz/panel/issues/4401"],
+    "reference/widgets/IntRangeSlider.ipynb": ["https://github.com/holoviz/panel/issues/4402"],
+    "reference/widgets/MultiChoice.ipynb": ["https://github.com/holoviz/panel/issues/4403"],
+    "reference/widgets/RangeSlider.ipynb": ["https://github.com/holoviz/panel/issues/4402"],
+    "reference/widgets/SpeechToText.ipynb": ["https://github.com/holoviz/panel/issues/4404"],
+    "reference/widgets/Terminal.ipynb": ["https://github.com/holoviz/panel/issues/4407"],
+    "gallery/components/VuePdbInput.ipynb": ["https://github.com/holoviz/panel/issues/4417"],
+}
+
+def _get_dependencies(nbpath: pathlib.Path):
+    key = str(nbpath).split("examples/")[-1]
+    dependencies = DEPENDENCIES.get(key, DEFAULT_DEPENDENCIES)
+    dependencies = [repr(d) for d in dependencies if not d in DEPENDENCY_NOT_IMPORTABLE]
+    return dependencies
+
+assert _get_dependencies("reference/panes/Folium.ipynb")==["'panel'", "'folium'"]
+
+def _to_source(dependencies):
+    source = f"import piplite\nawait piplite.install([{', '.join(dependencies)}])"
+
+    if "'pyodide-http'" in dependencies:
+        source += "\n\nimport pyodide_http\npyodide_http.patch_all()"
+
+    return source
```

Comment:
I don't think this is needed, pyodide-http gets auto-applied by Panel if it's importable.

```suggestion
    return f"import piplite\nawait piplite.install([{', '.join(dependencies)}])"
```